core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/services/impl/OptionSourceResolver.java

```diff
@@ -28,50 +30,58 @@
 import com.exadel.aem.toolkit.core.optionprovider.services.OptionProviderService;
 
 /**
- * Contains methods for resolving provided URIs (representing either JCR paths or HTTP endpoints) to option data sources
+ * Contains methods for resolving provided URIs (representing either JCR paths, HTTP endpoints or class name) to option data sources
  * @see OptionProviderService
  */
 interface OptionSourceResolver {
     OptionSourceResolver HTTP_RESOLVER = new HttpOptionSourceResolver();
     OptionSourceResolver JCR_RESOLVER = new JcrOptionSourceResolver();
+    OptionSourceResolver ENUM_RESOLVER = new EnumOptionSourceResolver();
+    String ID_PATTERN = "\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*";
 
     /**
-     * Called by {@link OptionSourceResolver#resolve(SlingHttpServletRequest, String, String)} to retrieve a {@link
-     * Resource} object representing the option datasource. Depending on the implementation, this method either
-     * retrieves a JCR entry or gets an HTTP response and converts it into a virtual resource
-     * @param request Current {@code SlingHttpServletRequest}
-     * @param uri     Either a JCR path to the {@code Resource} that represents a datasource or contains a link to the
-     *                actual datasource, or else an HTTP endpoint containing the datasource data in JSON format
-     * @return {@code Resource} instance, or null
+     * Method uses
+     * @param request        {@link SlingHttpServletRequest} and
+     * @param pathParameters {@link PathParameters} to
+     * @return {@link Resource} built using first basic {@link String} path
      */
-    Resource resolve(SlingHttpServletRequest request, String uri);
+    Resource pathResolve(SlingHttpServletRequest request, PathParameters pathParameters);
```

Comment:
Let me disagree with the distingushing between "pathResolve" and "fallbackResolve" methods.
Whether we are dealing with "path" and "fallbackPath", or else "class" and "fallbackClass", the procedure stays the same for the both cases. There is no chance that some "path" is processed in one way while "fallback path" is processed in another way. "Path" and "fallbackPath" are just two alternating arguments for the same routine. Therefore the "resolve" method must stay the same, or else it will be confusing.

Besides, "pathResolve" does not sound quite English) They tend to put object after the verb, and not before.